src/function/curry.ts

```diff
@@ -100,63 +99,15 @@ export function curry<P1, P2, P3, P4, R>(
  * }
  * const curriedFiveArgFunc = curry(fiveArgFunc);
  * const add1 = curriedFiveArgFunc(1);
- * const add2 = add1(2);
- * const add3 = add2(3);
- * const add4 = add3(4);
- * console.log(add4(5)); // 15
+ * const add3 = add1(2);
+ * const add6 = add3(3);
+ * const add10 = add6(4);
+ * console.log(add10(5)); // 15
  */
 export function curry<P1, P2, P3, P4, P5, R>(
   func: (p1: P1, p2: P2, p3: P3, p4: P4, p5: P5) => R
 ): (p1: P1) => (p2: P2) => (p3: P3) => (p4: P4) => (p5: P5) => R;
 
-/**
- * Curries a function, allowing it to be called with a single argument at a time and returning a new function that takes the next argument.
- * This process continues until all arguments have been provided, at which point the original function is called with all accumulated arguments.
- *
- * @param {(...args: any[]) => any} func - The function to curry.
- * @returns {(...args: any[]) => any} A curried function that can be called with a single argument at a time.
- *
- * @example
- * function sum(a: number, b: number, c: number) {
- *   return a + b + c;
- * }
- *
- * const curriedSum = curry(sum);
- *
- * // The parameter `a` should be given the value `10`.
- * const sum10 = curriedSum(10);
- *
- * // The parameter `b` should be given the value `15`.
- * const sum25 = sum10(15);
- *
- * // The parameter `c` should be given the value `5`. The function 'sum' has received all its arguments and will now return a value.
- * const result = sum25(5);
- */
-export function curry(func: (...args: any[]) => any): (...args: any[]) => any;
-
-/**
- * Curries a function, allowing it to be called with a single argument at a time and returning a new function that takes the next argument.
- * This process continues until all arguments have been provided, at which point the original function is called with all accumulated arguments.
- *
- * @param {(...args: any[]) => any} func - The function to curry.
- * @returns {(...args: any[]) => any} A curried function that can be called with a single argument at a time.
- *
- * @example
- * function sum(a: number, b: number, c: number) {
- *   return a + b + c;
- * }
- *
- * const curriedSum = curry(sum);
- *
- * // The parameter `a` should be given the value `10`.
- * const sum10 = curriedSum(10);
- *
- * // The parameter `b` should be given the value `15`.
- * const sum25 = sum10(15);
- *
- * // The parameter `c` should be given the value `5`. The function 'sum' has received all its arguments and will now return a value.
- * const result = sum25(5);
- */
```

Comment:
Since the overloaded signatures already have good JSDoc, we can delete this jsdoc of function implementation and keep only the necessary sections.